node/src/test/scala/com/wavesplatform/transaction/OrderSpecification.scala

```diff
@@ -43,36 +44,91 @@ class OrderSpecification extends PropSpec with ValidationMatcher with NTPTime {
   }
 
   property("Order expiration validation") {
-    forAll(arbitraryOrderGen) { order =>
-      val isValid = order.isValid(ntpTime.correctedTime())
-      val time    = ntpTime.correctedTime()
-      whenever(order.expiration < time || order.expiration > time + Order.MaxLiveTime) {
-        isValid shouldBe not(valid)
-      }
+    val versions = Seq(Order.V1, Order.V2, Order.V3, Order.V4)
+    val sender = TxHelpers.signer(1)
+    val matcher = TxHelpers.signer(2)
+    val pair = AssetPair(IssuedAsset(ByteStr.fill(32)(1)), IssuedAsset(ByteStr.fill(32)(2)))
+    val price = 1
+    val amount = 100
+    val time = 10000000000L
+    val expirationTime = 20000000000L
+    val matcherFee = 2
+
+    versions.foreach { version =>
+      val matcherFeeAsset = if (version == 3) IssuedAsset(ByteStr.fill(32)(3)) else Waves
+
+      Order.buy(version, sender, matcher.publicKey, pair, amount, price, time, expirationTime, matcherFee, matcherFeeAsset).explicitGet().isValid(30000000000L) shouldBe not(valid)
+      Order.buy(version, sender, matcher.publicKey, pair, amount, price, time, expirationTime, matcherFee, matcherFeeAsset).explicitGet().isValid(expirationTime - Order.MaxLiveTime - 1) shouldBe not(valid)
+
+      Order.sell(version, sender, matcher.publicKey, pair, amount, price, time, expirationTime, matcherFee, matcherFeeAsset).explicitGet().isValid(30000000000L) shouldBe not(valid)
+      Order.sell(version, sender, matcher.publicKey, pair, amount, price, time, expirationTime, matcherFee, matcherFeeAsset).explicitGet().isValid(expirationTime - Order.MaxLiveTime - 1) shouldBe not(valid)
```

Comment:
Why can't we create a variable and use it instead of copying such long expressions? Below too

Reply:
Done